### ISSUE TITLE
Neocomplete breaks autoindentation

### ISSUE THREAD

**Issue by apirogov** [Shougo/neocomplete.vim#311]
## Environment Information
- OS: Arch Linux
- Vim version: VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Oct 11 2014 03:20:17)  1-473
## Minimal vimrc less than 50 lines

```
let iCanHazVundle=1
let vundle_readme=expand('~/.vim/bundle/vundle/README.md')
if !filereadable(vundle_readme)
  echo "Installing Vundle.."
  echo ""
  silent !mkdir -p ~/.vim/bundle
  silent !git clone https://github.com/gmarik/vundle ~/.vim/bundle/vundle
  let iCanHazVundle=0
endif

filetype off                   " required!
set rtp+=~/.vim/bundle/vundle/
call vundle#rc()
" let Vundle manage Vundle, required!
Plugin 'gmarik/vundle'

Plugin 'Shougo/neocomplete'
let g:neocomplete#enable_at_startup = 1
let g:neocomplete#enable_smart_case = 1
let g:neocomplete#sources#syntax#min_keyword_length = 3
" Plugin key-mappings.
inoremap <expr><C-g>     neocomplete#undo_completion()
inoremap <expr><C-l>     neocomplete#complete_common_string()
" Recommended key-mappings.
" <CR>: close popup and save indent.
inoremap <silent> <CR> <C-r>=<SID>my_cr_function()<CR>
function! s:my_cr_function()
  return neocomplete#close_popup() . "\<CR>"
  " For no inserting <CR> key.
  "return pumvisible() ? neocomplete#close_popup() : "\<CR>"
endfunction
" <TAB>: completion.
inoremap <expr><TAB>  pumvisible() ? "\<C-n>" : "\<TAB>"
" <C-h>, <BS>: close popup and delete backword char.
inoremap <expr><C-h> neocomplete#smart_close_popup()."\<C-h>"
inoremap <expr><BS> neocomplete#smart_close_popup()."\<C-h>"
inoremap <expr><C-y>  neocomplete#close_popup()
inoremap <expr><C-e>  neocomplete#cancel_popup()

filetype plugin indent on " required!
if iCanHazVundle == 0
  echo "Installing Bundles, please ignore key map error messages"
  echo ""
  :BundleInstall
endif
```
## Reproducable ways from Vim starting
1. Open new file test.tex with following contents:

```
\begin{enumerate}
\textbf{test}
\end{enumerate}
```
1. Put your cursor at beginning of second line:

```
|\textbf{test}
```
1. Write: `\item` (with space at the end)

Expected output: (| = cursor)

```
\begin{enumerate}
        \item |\textbf{test}
\end{enumerate}
```

Actual output:

```
\begin{enumerate}
        \ite |m\textbf{test}
\end{enumerate}
```

So basically neocomplete somehow messes with the cursor position. This also happens with other filetypes, where indentation changes while typing, so I am quite sure it's not the tex filetype plugins fault...
